org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -2262,11 +2262,17 @@ public List<String> getCleanUpActionsOnSave() {
 	 * @return whether the options are changed or not
 	 */
 	public boolean updateAnnotationNullAnalysisOptions() {
+		boolean hasAnnotationNullAnalysisTypes = this.hasAnnotationNullAnalysisTypes();
+		if (hasAnnotationNullAnalysisTypes) {
+			Hashtable<String, String> javaCoreOptions = JavaCore.getOptions();
+			javaCoreOptions.put(JavaCore.COMPILER_PB_SYNTACTIC_NULL_ANALYSIS_FOR_FIELDS, JavaCore.ENABLED);
+			JavaCore.setOptions(javaCoreOptions);
+		}
 		switch (this.getNullAnalysisMode()) {
 			case automatic:
 				return this.updateAnnotationNullAnalysisOptions(true);
 			case interactive:
-				if (this.hasAnnotationNullAnalysisTypes()) {
+				if (hasAnnotationNullAnalysisTypes) {
```

Comment:
This is the right idea, but not here. Look at the method that's called in both the `automatic` and `default` case. You'll see a method that sets many different options.